### DISCUSSION TITLE
Help with playing a local mp3

### DISCUSSION THREAD

**Discussion by hidara2000** [flet-dev/flet#2865] [General]
Hi, 
I have this web app that I'm playing with and can:

- stream an mp3 from a url
- Show an image from the assets folder

However, I can't seem to figure out how to play a local mp3 in the assets folder. The following code shows the image from the assets folder but I hear no audio. If replace the path with a web url all works perfectly.

any help would be appreciated :)

``` python
import flet as ft


def main(page: ft.Page):
    audio1 = ft.Audio("sample.mp3", autoplay=True)
    page.overlay.append(audio1)
    page.add(
        ft.Text("This is an app with background audio."),
        ft.ElevatedButton("Stop playing", on_click=lambda _: audio1.pause()),
        ft.ElevatedButton("Start playing", on_click=lambda _: audio1.play()),
        ft.Image("icon.png"),
    )


ft.app(target=main, view=ft.AppView.WEB_BROWSER, port=8550, assets_dir="assets")

```

![image](https://github.com/flet-dev/flet/assets/15170494/80e77e66-294b-4a05-8ad0-6fe692ca1872)

**Comment by liudonghua123** [comment]
I go through the doc https://flet.dev/docs/controls/audio/#src and https://flet.dev/docs/controls/image#src.

Maybe you should prefix the filename with a `/`. Change `ft.Audio("sample.mp3", autoplay=True)` to `ft.Audio("/sample.mp3", autoplay=True)`.

**Reply by hidara2000** [reply]
Cheers but tried both and no luck. Replaced the audio widget with a Video one and have no problems playing a local mp4

**Comment by ndonkoHenri** [comment]
Can you share the file you are using? It might be damaged or not have the correct extension,
On my end, I added [this](https://luan.xyz/files/audio/ambient_c_motion.mp3) audio file into my assets, then tried and it worked... (both with the leading / and without).

**Reply by hidara2000** [reply]
appreciated. I downloaded the MP3 above and saved it to the asset directory. Still no sound but at the same time can play directly from the web url.

**Comment by hidara2000** [comment]
This is a bit strange. Sometimes it works out of the blue. I'm using Edge on both Android and Windows and the docs say that autoplay doesn't work. I took this as meaning that I have to manually click a button to start it. 
I'm thinking:

- Since it's not working on both Windows and Android the problem is either with Edge but probably not plugins or antivirus.
- When I first start it in Edge it gives me the option to download as a web app. When I do this and run the web app it works as expected.

Stumped!
